android-components/components/feature/fxsuggest/src/main/java/mozilla/components/feature/fxsuggest/facts/FxSuggestFactsMiddleware.kt

```diff
@@ -40,12 +42,14 @@ class FxSuggestFactsMiddleware : Middleware<BrowserState, BrowserAction> {
         context: MiddlewareContext<BrowserState, BrowserAction>,
         action: BrowserAction,
     ) = when (action) {
-        is AwesomeBarAction.EngagementFinished ->
-            if (action.abandoned) Unit else emitSuggestionFacts(context.state.awesomeBarState)
+        is AwesomeBarAction.EngagementFinished -> emitSuggestionFacts(
```

Comment:
This action is dispatched when the user finishes interacting with the list of results. `action.abandoned` is `true` if the user abandoned their search by tapping away to dismiss the awesomebar, or `false` if the user completed their search (an "engagement") by tapping on a suggestion, or typing a URL or a search query and tapping "enter".

Previously, we ignored abandonments, because our partners don't count those as impressions. But our Desktop analyses _do_ count impressions for abandonments, so we're now emitting those facts, with the `engagementAbandoned` flag. Fenix's `MetricController` figures out which events to record and pings to send based on that flag.